Update links in metadata section on single result page

obulat opened the pull request:
<!-- prettier-ignore -->
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, please consider opening one before creating this pull request. -->

Fixes #2731 by @obulat

## Description
<!-- Concisely describe what the pull request does. -->
<!-- Add screenshots, videos, or other media to show the problem and the solution when appropriate. -->
This PR updates the metadata section of the single result page.

The metadata items match the updates styles: [Metadata component in Figma](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?type=design&node-id=5171-16563&mode=design&t=blxAwawVa4wBfeT4-4), [Metadata in the audio page in Figma](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?type=design&node-id=5645-29968&mode=design&t=wlOZdKnd7mwKmcQa-4)

The items don't wrap anymore, unless they don't fit. You can see that in the snapshot comparisons: previously, the long values would be split between 2-3 lines, and now they are all on one line.

The external links to the source, the provider and the album have an "external link" icon added to them.

The "source" link is updated to link to the source's landing page instead of the media's foreign landing URL.


## Testing Instructions
<!-- Give steps for the reviewer to verify that this PR fixes the problem; or delete this section entirely. -->
Check that the single result page metadata section matches the mockups. The code should make sense, too.
When you click on the links, correct analytics events should be fired, and only one per link (not external link click + some other event).

_Note:_ the (i) information popover explaining the difference between provider and source is not in scope of this PR

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. -->

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

obulat commented on frontend/src/types/analytics.ts:
```diff
@@ -216,6 +216,18 @@ export type Events = {
     url: string
   }
+  /**
+   * Description: The user visits a source's link in the single result UI
```
This event is for the external link to the source. Here, we can also get the id of the single result from which the link was clicked. However, for other source links (on the source collection page), there is no single media. Do we want to use the `id` as an optional prop, or is it not necessary?

obulat commented on frontend/src/types/analytics.ts:
```diff
@@ -216,6 +216,18 @@ export type Events = {
+   */
+  VISIT_SOURCE_LINK: {
+    /** The unique ID of the media */
+    id: string
```
```suggestion
    id?: string
```

fcoveram reviewed (commented):
The change looks great.

### Focus ring

I just noticed that the UI's side areas crop the focus ring. See how subtle the component corners render parts of the ring.

<img width="94" alt="Metadata component with glitchy pixels on all corners" src="https://github.com/WordPress/openverse/assets/895819/39b84027-8dcc-4ae1-9fe0-6ae096ef15fb">

I wonder if there is a site-wide solution that keeps the spacing values and sets the focus ring above all UI layers. Other parts of the interface face the same issue.

### Section layout

On the other hand, the new col-gap spacing in [this snapshot](https://github.com/WordPress/openverse/pull/2851/files#diff-639e5d75aab15327df7c26711e5fbca1c0d35087e3f8d803e888c869e83457aa) looks great, but [here](https://github.com/WordPress/openverse/pull/2851/files#diff-22550982884395bcd893fc6efdf85b4ff836adbd7f48dc652e0b7be40081671c) it looks unbalanced due to the album's name length.

The solution implemented works well, but to display a balanced layout we can try a max-width of `160px (w-20)` in the component, and place them in columns where items in second row follow the width of same item in first row. Like the quick mockups atttached below. Otherwise, item in second row will look off the grid, like [this snapshot](https://github.com/WordPress/openverse/pull/2851/files#diff-188bf15db0d9751637cc41b7cdf8b70d9bc39254352972d28d5848b62cc919ef).

<img width="1280" alt="Metada components in media information section. Guidelines are visible in the component borders" src="https://github.com/WordPress/openverse/assets/895819/985a3a35-7a5f-4bc3-b17d-9184d12ada77">

fcoveram commented:
I forgot to add that on `xs / sm`, the category's label width (left column) could be the same as the longest label (Dimensions, as in the example attached below) to avoid the big gap between the two columns.

<img width="569" alt="Metadata section on mobile, CSS values of category field are highlighted" src="https://github.com/WordPress/openverse/assets/895819/380d1230-6566-42a7-9cc6-a737a5472a82">

obulat commented on frontend/src/pages/audio/_id/index.vue:
```diff
@@ -138,21 +138,3 @@ export default defineComponent({
   head: {},
 })
 </script>
```
These styles were not used: we don't have "audio-page" class anymore.

fcoveram reviewed (commented):
The horizontal layout looks correct, but focus ring is still cropped by the content.

Not sure if the suggestion [I shared](https://user-images.githubusercontent.com/895819/263685555-985a3a35-7a5f-4bc3-b17d-9184d12ada77.png) above is possible, but I found [this example](https://developer.mozilla.org/en-US/docs/Web/CSS/fit-content_function) on MDN Web Docs that seems to address it.

AetherUnbound commented:
![image](https://github.com/WordPress/openverse/assets/10214785/a710850a-baeb-430a-b0fc-15e8ff31134a)

Is it intentional that the metadata headings & values look a little staggered horizontally in the small view? This is on Firefox with the iPhone 11 viewport size.

fcoveram commented:
> Is it intentional that the metadata headings & values look a little staggered horizontally in the small view? This is on Firefox with the iPhone 11 viewport size.

Yes. It's part of the design

obulat commented:
@fcoveram, I had to add a small padding in the `xs` size to get the focus rings to show. Let me know what you think

AetherUnbound commented:
Sorry to be clear:
![image](https://github.com/WordPress/openverse/assets/10214785/1ac1be19-4660-4f68-89f5-e4082f34d8a4)
It looks like "Dimensions" is aligned to the top of the box I've drawn, where as "1024 x 819 pixels" is aligned to the bottom. That's intended behavior?

obulat commented:
> Is it intentional that the metadata headings & values look a little staggered horizontally in the small view? This is on Firefox with the iPhone 11 viewport size.

Thank you for adding the snapshots, @AetherUnbound. The misalignment of items was caused by the padding I added to the links to make sure that they show the focus ring when you focus them with the keyboard (without the padding, they were getting cut off). I've also added the padding to the titles, so now the titles should be aligned with the items, as you can see in the `xs` snapshots in this PR.

fcoveram reviewed (dismissed):
The change looks great. The padding is a good solution and is worth considering the same approach when implementing the tag component to reach the same layout spacing.

Regarding changes, and since the padding increases the gap between elements, I think we can change the following:
* On `xs` change the gap between rows from `24px` to `8px`.
* On `sm` and beyond
   * Change the gap between columns from `40px` to `32px`.
   * Change the padding-bottom of category field from `8px` to `4px`.

dhruvkb reviewed (commented):
From the screenshots, it looks like the genre is not cased properly. What used to be 'Titlecase' is now all 'lowercase'.

AetherUnbound reviewed (approved):
LGTM, thanks for correcting the padding on the smaller sizes!
